src/core/thread/mesh_forwarder.cpp

```diff
@@ -2223,6 +2239,7 @@ void MeshForwarder::LogIp6Message(MessageAction aAction, const Message &aMessage
 
     case kMessageDrop:
         actionText = "Dropping";
+        mIpCounters.mRxFailure++;
```

Comment:
With this, `mRxFailure` tracks number of larger IP messages (requiring fragments) where we did receive the first fragment but not necessarily all rest (thus dropping it from reassembly queue).... So it won't include failed smaller received IP6 messages... but I am not sure if we can do any better to track that.